### PR TITLE
need pull_request trigger for branch updates

### PR DESCRIPTION
**Pull request by jakecoffman**
#6155 removed the pull_request trigger, but we need it since it triggers when we update a branch from master, for instance, or push a small change. That way we don't need to ask for reapproval in those cases. 

I **think** types: edited is the correct thing to put. We don't need the workflow to run when the PR opens, or closes, etc.

I also fixed the names of the latest release workflow being the same as the branch release workflow, which then got me carried away to make all the names shorter and more accurate.

### CODE REVIEW

File: .github/workflows/docker-branch-releases.yml
Comment (jakecoffman) [inline .github/workflows/docker-branch-releases.yml:13]: This name was from a previous iteration where we uploaded to build artifacts.

**Review by deivid-rodriguez — APPROVED** [review]

**Comment by jakecoffman** [comment]
I'll merge this after @pavera's merge so we don't have to update branches excessively.

**Pull request merged**
